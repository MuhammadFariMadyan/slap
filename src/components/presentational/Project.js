import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { format } from "date-fns";
import "typeface-roboto";

import Thumbnail from "./Thumbnail";
import SmallButton from "./SmallButton";

const Container = styled.div`
  margin-bottom: 16px;
`;

export const Name = styled.h2`
  font-size: 24px;
  font-weight: bold;
  font-family: Roboto;
  color: #333;
  margin-top: 12px;
  margin-bottom: 4px;
  cursor: pointer;
`;

const Date = styled.p`
  font-size: 18px;
  font-family: Roboto;
  color: #aaa;
  margin: 0px;
  margin-bottom: 12px;
`;

const Project = ({ name, timestamp, contents, onClick, onClickDelete }) => (
  <Container>
    <Thumbnail contents={contents} onClick={onClick} />
    <Name onClick={onClick}>{name}</Name>
    <Date>{format(timestamp, "MMM DD, YYYY")}</Date>
    <SmallButton caption="Delete" onClick={onClickDelete} />
  </Container>
);

Project.propTypes = {
  name: PropTypes.string,
  timestamp: PropTypes.number,
  contents: PropTypes.array,
  onClick: PropTypes.func,
  onClickDelete: PropTypes.func
};

export default Project;
