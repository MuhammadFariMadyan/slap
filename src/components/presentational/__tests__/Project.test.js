import React from "react";
import { shallow } from "enzyme";
import "jest-styled-components";

import Project, { Name } from "../Project";

describe("Project", () => {
  jest.mock("../Thumbnail", () => "Thumbnail");
  jest.mock("../SmallButton", () => "SmallButton");

  const setup = propOverrides => {
    const props = {
      name: "My Own Slides",
      timestamp: 1537004755445,
      contents: [
        {
          id: "content-1",
          slideID: "slide-1",
          projectID: "project-1",
          type: "text",
          content: "Hello World",
          position: {
            x: 32,
            y: 32
          }
        },
        {
          id: "content-2",
          slideID: "slide-1",
          projectID: "project-1",
          type: "text",
          content: "Stay hungry",
          position: {
            x: 128,
            y: 96
          }
        }
      ],
      onClick: jest.fn(),
      onClickDelete: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<Project {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it("should calls 'onClick' callback while name clicked", () => {
    const { wrapper, props } = setup();

    wrapper.find(Name).simulate("click");

    expect(props.onClick).toBeCalled();
  });

  it("should calls 'onClickDelete' callback while delete clicked", () => {
    const { wrapper, props } = setup();

    wrapper.find("SmallButton").simulate("click");

    expect(props.onClickDelete).toBeCalled();
  });
});
