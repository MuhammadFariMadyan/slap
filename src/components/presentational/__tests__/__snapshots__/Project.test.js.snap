// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project should renders default 1`] = `
<styled.div>
  <Thumbnail
    contents={
      Array [
        Object {
          "content": "Hello World",
          "id": "content-1",
          "position": Object {
            "x": 32,
            "y": 32,
          },
          "projectID": "project-1",
          "slideID": "slide-1",
          "type": "text",
        },
        Object {
          "content": "Stay hungry",
          "id": "content-2",
          "position": Object {
            "x": 128,
            "y": 96,
          },
          "projectID": "project-1",
          "slideID": "slide-1",
          "type": "text",
        },
      ]
    }
    onClick={[Function]}
    width={240}
  />
  <styled.h2
    onClick={[Function]}
  >
    My Own Slides
  </styled.h2>
  <styled.p>
    Sep 15, 2018
  </styled.p>
  <SmallButton
    caption="Delete"
    onClick={[Function]}
  />
</styled.div>
`;
